src/snowflake/snowpark/functions.py

```diff
@@ -2935,6 +2945,39 @@ def concat_ws(*cols: ColumnOrName) -> Column:
     return builtin("concat_ws")(*columns)
 
 
+def concat_ws_ignore_nulls(sep: str, *cols: ColumnOrName) -> Column:
+    """Concatenates two or more strings, or concatenates two or more binary values. Null values are ignored.
+
+    Args:
+        sep: The separator to use between the strings.
+
+    Examples::
+        >>> df = session.create_dataframe([
+        ...     ['Hello', 'World', None],
+        ...     [None, None, None],
+        ...     ['Hello', None, None],
+        ... ], schema=['a', 'b', 'c'])
+        >>> df.select(concat_ws_ignore_nulls(',', df.a, df.b, df.c)).show()
+        ----------------------------------------------------
+        |"CONCAT_WS_IGNORE_NULLS(',', ""A"",""B"",""C"")"  |
+        ----------------------------------------------------
+        |Hello,World                                       |
+        |                                                  |
+        |Hello                                             |
+        ----------------------------------------------------
+        <BLANKLINE>
+    """
+    columns = [_to_col_if_str(c, "concat_ws_ignore_nulls") for c in cols]
+    names = ",".join([c.get_name() for c in columns])
+    input_column_array = array_construct_compact(*columns)
+    reduced_result = builtin("reduce")(
```

Comment:
reduce is to return a single value. Should it be filter?

Reply:
Another way is to use `concat_was(filter(...))` to remove the nulls first then concat.

Reply:
Return value of `filter` is an array. `concat_ws` does not take `ARRAY` as an input therefore we had to use reduce.